Libraries/DirectMLX.h

```diff
@@ -2004,7 +2004,11 @@ namespace dml
     // ---------------------------------------------------------------------------------------------------------------
 
     // If `axes` is not specified, by default this reduces the entire tensor to single element.
-    inline Expression Reduce(Expression input, DML_REDUCE_FUNCTION function, Span<const uint32_t> axes = {})
+    inline Expression Reduce(
+        Expression input, 
+        DML_REDUCE_FUNCTION function, 
+        DML_TENSOR_DATA_TYPE targetDataType = DML_TENSOR_DATA_TYPE_UNKNOWN, 
```

Comment:
I think it would be cleaner if this simply defaulted to UINT32, then you could remove the conditional below on line 2050.

Reply:
🤔 It would be shorter, but isn't a weird to favor one reduction as the default value above all other reduction types? If ReduceProd for some reason had a different output data type, what would we choose? UNKNOWN makes sense to me here, allowing any logic specific to ArgMax/Min to be isolated.

Reply:
And it's bad practice to require that an `if` statement *override* a default for the common case. You want the common case to be the straight-through flow, and the `if` statements to be the branchy special thing.

Reply:
I think the common case would have to be in an `if` statement regardless since it's assigned to `inputTensor.dataType`, but it makes sense that favoring one reduction isn't ideal. @adtsai Thoughts?

Reply:
If someone explicitly passes an `outputDataType`, we don't want to override it, right? (even if it's not equal to the input, because dmlx should be a mostly thin pass-through, and this makes us more robust to future changes). How about:

```
        if (outputDataType = DML_TENSOR_DATA_TYPE_UNKNOWN) 
        {
            if (function == DML_REDUCE_FUNCTION_ARGMIN || function == DML_REDUCE_FUNCTION_ARGMAX)
            {
                // Default to UINT32 if the output type wasn't specified
                outputDataType = DML_TENSOR_DATA_TYPE_UINT32;
            }
            else
            {
                outputDataType = inputTensor.dataType;
            }
        }
```